dist/s_test_suite_no_executable

```diff
@@ -5,9 +5,9 @@ set -euf -o pipefail
 # All test scripts should be routed through "run.py", running them directly is an anti-pattern.
 # Use the executable bits on the scripts as weak evidence of the anti-pattern.
 workspace="$(git rev-parse --show-toplevel)"
-exes=$(find "$workspace"/test/suite -type f \( -perm -u+x -o -perm -g+x -o -perm -o+x \) ! -name 'run.py' -print)
+exes=$(find "$workspace"/test/suite -type f -perm +X ! -name 'run.py' -print)
```

Comment:
nit: `-print` is superfluous here - it's the default

Reply:
oops.. `+X` should be `+111`

Reply:
The method you have specified does not work, see output below, I also tried `+0111` (octal) but that has the same result as `+111`. Also `+X` does not work.

Note: I want to match if executable mode bits are set.

```
ip-10-0-101-50:dist [1]$ cat s_test_suite_no_executable
#!/usr/bin/env bash

set -euf -o pipefail

# All test scripts should be routed through "run.py", running them directly is an anti-pattern.
# Use the executable bits on the scripts as weak evidence of the anti-pattern.
workspace="$(git rev-parse --show-toplevel)"
#exes=$(find "$workspace"/test/suite -type f \( -perm -u+x -o -perm -g+x -o -perm -o+x \) ! -name 'run.py')
#exes=$(find "$workspace"/test/suite -type f \( -perm -u+x -o -perm -g+x -o -perm -o+x \) ! -name 'run.py')
exes=$(find "$workspace"/test/suite -type f -perm +111 ! -name 'run.py')
if [[ -n "$exes" ]]; then
    echo "The following files should not be executable:"
    echo "$exes"
    exit 1
fi

exit 0
ip-10-0-101-50:dist [0]$ ./s_test_suite_no_executable
find: invalid mode '+111'
ip-10-0-101-50:dist [1]$ cat s_test_suite_no_executable
#!/usr/bin/env bash

set -euf -o pipefail

# All test scripts should be routed through "run.py", running them directly is an anti-pattern.
# Use the executable bits on the scripts as weak evidence of the anti-pattern.
workspace="$(git rev-parse --show-toplevel)"
#exes=$(find "$workspace"/test/suite -type f \( -perm -u+x -o -perm -g+x -o -perm -o+x \) ! -name 'run.py')
exes=$(find "$workspace"/test/suite -type f -perm +X ! -name 'run.py')
#exes=$(find "$workspace"/test/suite -type f -perm +111 ! -name 'run.py')
if [[ -n "$exes" ]]; then
    echo "The following files should not be executable:"
    echo "$exes"
    exit 1
fi

exit 0
ip-10-0-101-50:dist [0]$ ./s_test_suite_no_executable
ip-10-0-101-50:dist [0]$ cat s_test_suite_no_executable
#!/usr/bin/env bash

set -euf -o pipefail

# All test scripts should be routed through "run.py", running them directly is an anti-pattern.
# Use the executable bits on the scripts as weak evidence of the anti-pattern.
workspace="$(git rev-parse --show-toplevel)"
exes=$(find "$workspace"/test/suite -type f \( -perm -u+x -o -perm -g+x -o -perm -o+x \) ! -name 'run.py')
#exes=$(find "$workspace"/test/suite -type f -perm +X ! -name 'run.py')
#exes=$(find "$workspace"/test/suite -type f -perm +111 ! -name 'run.py')
if [[ -n "$exes" ]]; then
    echo "The following files should not be executable:"
    echo "$exes"
    exit 1
fi

exit 0
ip-10-0-101-50:dist [0]$ ./s_test_suite_no_executable
The following files should not be executable:
/home/ubuntu/src/wiredtiger/test/suite/test_cursor23.py
/home/ubuntu/src/wiredtiger/test/suite/test_cursor_bound01.py
/home/ubuntu/src/wiredtiger/test/suite/test_cursor_bound03.py
/home/ubuntu/src/wiredtiger/test/suite/test_cursor_bound09.py
/home/ubuntu/src/wiredtiger/test/suite/test_cursor_bound10.py
/home/ubuntu/src/wiredtiger/test/suite/test_cursor_bound14.py
/home/ubuntu/src/wiredtiger/test/suite/test_cursor_bound12.py
/home/ubuntu/src/wiredtiger/test/suite/test_cursor_bound06.py
/home/ubuntu/src/wiredtiger/test/suite/test_cursor_bound07.py
```

Reply:
Hmm.. Let' catch up offline for this. `+111` seems to work on a Mac:
```
$ ls -l
total 0
----------  1 y.ershov  staff  0 11 Jan 06:35 file-000
---------x  1 y.ershov  staff  0 11 Jan 06:35 file-001
------x---  1 y.ershov  staff  0 11 Jan 06:35 file-010
------x--x  1 y.ershov  staff  0 11 Jan 06:35 file-011
---x------  1 y.ershov  staff  0 11 Jan 06:35 file-100
---x-----x  1 y.ershov  staff  0 11 Jan 06:35 file-101
---x--x---  1 y.ershov  staff  0 11 Jan 06:35 file-110
---x--x--x  1 y.ershov  staff  0 11 Jan 06:35 file-111
$ find . -type f -perm +111
./file-101
./file-100
./file-010
./file-011
./file-001
./file-110
./file-111
$ find . -type f -perm 111
./file-111
$ find . -type f -perm -111
./file-111
$ find . -type f -perm +x
./file-111
$ find . -type f -perm +a+x
./file-101
./file-100
./file-010
./file-011
./file-001
./file-110
./file-111
$ find . -type f -perm -a+x
./file-111
$ find . -type f -perm a+x
./file-111
```

Reply:
Ubuntu 22.04
```
ip-10-0-101-50:files [0]$ ls -l
total 0
-rwxrw-r-- 1 ubuntu ubuntu 0 Jan 10 22:50 f1
-rw-rwxr-- 1 ubuntu ubuntu 0 Jan 10 22:50 f2
-rw-rw-r-x 1 ubuntu ubuntu 0 Jan 10 22:50 f3
-rwxrwxr-x 1 ubuntu ubuntu 0 Jan 10 22:51 f4
ip-10-0-101-50:files [0]$ find -type f -perm -111
./f4
ip-10-0-101-50:files [0]$
```